### ISSUE TITLE
get node not working properly

### ISSUE THREAD

**Issue by DrZeta360** [O-Hahn/node-red-contrib-cos#21]
Hi,
I've been trying to use the get node but there seems be a few issues. If I leave the path and filename override blank, it will download the file, but only 15bytes out of nearly 3Mb. If I override to a different path, then it won't download at all.
I'm running it on a RPi 3B+ and the put node works well, so it's not the configuration.
Any ideas?
Thanks.

**Comment by O-Hahn** [comment]
this implementation does not support multipart download - but normally the max size of one objects is 100MB via http body... so this should not be a problem. 
The path to the object given must exist - be aware of the underlying os path (linux vs windows).

**Comment by DrZeta360** [comment]
Hi thanks for getting back to me.
The file is definitely below 100mb. The path exists as I am copying other files to and from it, and it should be fine as it is the same format I use for the put node, which works well.
I tried a few different paths and combinations and nothing seemed to work.
The stranger thing is that even with the default path I only get 15 bytes out of the 3mb.

**Comment by mek790** [comment]
Hi, I have the same issue as @DrZeta360.  

The "put", "del" and "qry" nodes all work fine but using the same files, buckets and APIs etc the "get" node returns only [object Object] and is 15bytes.  Did you find a solution to this?

Thanks for you help

**Comment by O-Hahn** [comment]
Thanks to @mek790 - issue is raised by setting some values in the msg.headers before. Therefore it overrides the needed internal token for accessing COS. See https://github.com/O-Hahn/node-red-contrib-cos/issues/22

**Comment by mek790** [comment]
Thank you for getting back to me so quickly.

I cannot completely understand the reply so I was wondering if you can help further.

My aim is to be able to "put" and "get" files with different file extensions (.txt, .png etc.) on / from the cloud using these nodes

1 - I am able to use the "put" node without any issues
2 - I am able to "get" the content of a .txt file and output it using the "msg.payload" Mode
3 - I am unable to "get" the "object" (file) complete from the bucket using the "file" Mode however, a new file with the "File Name" does appear in the "File Path" but its content is "[object Object]" (if I use a .txt file) and only 15bytes.

You mention "msg.header".  Are you able to explain what I need to do with this to allow me to "get" the entire file (i.e. download the entire .png file?

Thank you very much.  Sorry for not understanding, I am still learning.

**Comment by Oktober31** [comment]
Hello,
I have the same problem with files I send and that I tried to download. The file downloaded have 15bytes. But when I look in the cloud, they say that I downloaded the complete file. When I download with the URL, the file is good.

I think, it is a problem encoding, but I'm not sure. In the getObject(params = {}, callback), it is possible to send which encoding it is. Perhaps it is possible to send with msg.header, but I don't find the syntax.

Have you the answser ?

Thank you.
